### PR TITLE
Rendering Dataset Metadata

### PR DESCRIPTION
**Pull request by Omkar20895**
I have appended the datasetId, shortName and format fields in the method to the url. 
Please check it and if it looks good, we can replicate the same code in the rest of the methods. 
Thanks.

### CODE REVIEW

File: podaac/podaac.py [hunk 1]
```diff
@@ -36,12 +36,13 @@ def load_dataset_md(datasetId, shortName, format='iso'):
 
 	'''
 
-	url = 'http://podaac.jpl.nasa.gov/ws/metadata/dataset'
+	url = 'http://podaac.jpl.nasa.gov/ws/metadata/dataset/?datasetId='+datasetId+'&shortName='+shortName+'&format='+format
 	url += '?checkers={}&url-upload={}&response={}'
```
Comment (lewismc) [inline podaac/podaac.py]: This needs to come out

File: podaac/podaac.py [hunk 2]
```diff
@@ -36,12 +36,13 @@ def load_dataset_md(datasetId, shortName, format='iso'):
...
 	url = url.format(checkers, url_upload, response)
```
Comment (lewismc) [inline podaac/podaac.py]: So does this

**Comment by lewismc** [comment]
Thanks @Omkar20895 , yes please, if you can go ahead and populate the remainder of the functions that would be absolutely brilliant.
Thanks

**Comment by Omkar20895** [comment]
Okay sure!

**Comment by Omkar20895** [comment]
Hello @lewismc when I render the image in the load_image_granule function, do I need to create and save the image in the same folder or do I need to return it in a variable??

**Comment by lewismc** [comment]
In all honesty as this API is very young I am not 100% sure.
I think that this particular function may my be used as much as others
anyway.
Right now I would just capture it as a variable and the user can the.
Determine how to save the image. I think that this is described in the
documentation under the returns notation.
Thanks

On Saturday, April 16, 2016, Omkar <notifications@github.com> wrote:

> Hello @lewismc <https://github.com/lewismc> when I render the image in
> the load_image_granule function, do I need to create and save the image in
> the same folder or do I need to return it in a variable??
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/lewismc/podaacpy/pull/2#issuecomment-210852557>
>


-- 
*Lewis*

**Comment by Omkar20895** [comment]
I am unable to render the .nc file in the extract_granule, when I print the status of the response I get a '404' return code(page not found). I have tried to remove it and cross checked the url many times but couldn't resolve it. @lewismc can you please help me out here??

**Comment by lewismc** [comment]
Can you send an intermediate pull request so I can see your code?
What does the request look like?

On Saturday, April 16, 2016, Omkar <notifications@github.com> wrote:

> I am unable to render the .nc file in the extract_granule, when I print
> the status of the response I get a '404' return code(page not found). I
> have tried to remove it and cross checked the url many times but couldn't
> resolve it. @lewismc <https://github.com/lewismc> can you please help me
> out here??
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/lewismc/podaacpy/pull/2#issuecomment-210872292>
>


-- 
*Lewis*

**Comment by Omkar20895** [comment]
Look at my latest commit and you will find the request.

**Comment by lewismc** [comment]
The issue may be here 
```'&style'+style+'```
should this not be
```'&style='+style+'```

File: podaac/podaac.py [hunk 3]
```diff
@@ -381,7 +391,12 @@ def extract_granule(datasetId, shortName, granuleName, bbox, format):
 
 	'''
 
-	url = 'http://podaac.jpl.nasa.gov/ws/extract/granule'
+	url = 'http://podaac.jpl.nasa.gov/ws/extract/granule/?datasetId='+datasetId+'&shortName='+shortName+'&granuleName='+granuleName+'&bbox='+bbox+'&format='+format
+	granule = requests.get(url)
+	return granule
+
+result = extract_granule(shortName='ASCATA-L2-25km', granuleName='ascat_20130719_230600_metopa_35024_eps_o_250_2200_ovw.l2.nc', bbox='45,0,180,90', format='netcdf')
```
Comment (lewismc) [inline podaac/podaac.py]: Formatting is not correct here.

**Comment by Omkar20895** [comment]
That's an error in the load_image_granule, thanks for pointing it out. But, the page not found error was in the extract_granule method.

**Comment by Omkar20895** [comment]
I hardcoded it just so that you can test it, I thought of removing it after debugging the error.

**Comment by lewismc** [comment]
I checked the service and it turns out that a granule should be returned if
1) EITHER a datasetId OR shortName are provided, OR
2) if BOTH are provided
Basically, the request cannot contain an empty/null value for one or both of the values within the request.

**Comment by lewismc** [comment]
Additionally, it might be worth for us to consider error handling for each response if something goes wrong.

**Comment by Omkar20895** [comment]
Yeah we definitely need to consider it. I will update the pull request with 'try' and 'except' blocks of 'requests' module. Thanks

**Comment by Omkar20895** [comment]
Hello @lewismc I have a doubt, if there is an error then do I need to return void (or) do I need to print the error status eg : "HTTP: page not found" and then return void?? Thanks

**Comment by lewismc** [comment]
Returning a message along with an error trace is always helpful.

On Wednesday, April 20, 2016, Omkar <notifications@github.com> wrote:

> Hello @lewismc <https://github.com/lewismc> I have a doubt, if there is
> an error then do I need to return void (or) do I need to print the error
> status eg : "HTTP: page not found" and then return void?? Thanks
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/lewismc/podaacpy/pull/2#issuecomment-212423720>
>


-- 
*Lewis*

**Comment by Omkar20895** [comment]
@lewismc we need to update the docs since we have added error handling and the user needs to know that the functions might some times return error log instead of actual information.

**Comment by lewismc** [comment]
Or error handling, it does not matter at this stage. If we work with error
handling right now then that is fine. We can look to provide better logging
at a later stage.

On Wednesday, April 20, 2016, Omkar <notifications@github.com> wrote:

> @lewismc <https://github.com/lewismc> we need to update the docs since we
> have added error handling and the user needs to know that the functions
> might some times return error log instead of actual information.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/lewismc/podaacpy/pull/2#issuecomment-212507145>
>


-- 
*Lewis*

**Comment by lewismc** [comment]
Is this PR finished @Omkar20895 ?

**Comment by Omkar20895** [comment]
No not yet, few small changes remaining in the code.. :)

**Comment by lewismc** [comment]
@Omkar20895 what is still remaining to be done here? Please be explicit in your response as it is taking quite a while to finish these functions off. This is simple code and there are MUCH more complex things to get on to. Can you also send nose tests for each and every function call as we discussed at ApacheCon? Thanks

**Comment by Omkar20895** [comment]
The issue that I have raised about writing into .nc file is the only thing that is left out in this PR. I am writing nose tests, sorry for the delay.

**Comment by Omkar20895** [comment]
@lewismc For the web services shall I store the XML generated from the web app in a .xml file and compare it then with the XML returned by the function call??

**Comment by Omkar20895** [comment]
Or I can scrape the example xml from the website and then compare it with the one that has been obtained by the functional call.

**Comment by Omkar20895** [comment]
@lewismc Shall we close this pull request and look at the issue of writing into .nc file later on??

**Comment by lewismc** [comment]
Yes please commit the code thanks

On Monday, May 23, 2016, Omkar <notifications@github.com> wrote:

> @lewismc <https://github.com/lewismc> Shall we close this pull request
> and look at the issue of writing into .nc file later on??
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/lewismc/podaacpy/pull/2#issuecomment-220899047>
>


-- 
*Lewis*

**Pull request merged**